examples/lib/versionManagement.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Functions for getting version information.
+ * Note: these functions are not used by the service worker to check for updates;
+ * the service worker updates whenever sw.js has changed.
+ * sw.js is changed when grunt replace:version is run. This task is run during
+ * grunt build, serve, and productions tasks.
+ */
+
+/**
+ * Get the current version number from package.json on the homepage.
+ * @param {function} callback The function that uses the version number.
+ */
+function getLatestVersionNumber(callback) {
+  // Get the homepage reference from the local package.json.
+  var homepage = require('../../package.json').homepage;
+  var request = new XMLHttpRequest();
+  request.onreadystatechange = function() {
+    if(request.readyState == 4 && request.status == 200) {
+      var response = JSON.parse(this.responseText);
+      var latestVersionNumber = response.version;
+
+      // Do something with the version number using a callback function.
+      callback(latestVersionNumber);
+    }
+  }
+
+  // Get the package.json file from online using a GET request.
+  request.open("GET", homepage + "/package.json", true);
+  request.send();
+}
+
+// Get the version number from the local package.json file.
+function getLocalVersionNumber() {
+  return require('../../package.json').version;
```

Comment:
Why does the require statement appear twice?

Reply:
Also, I think adding all the require statements at the start of the file is a good practice.

Reply:
I moved the require statement to the start of the file, so it's now called only once.